File: Examples/test-suite/go/template_typedef_import_runme.go
package main

import "template_typedef_cplx2"
import "template_typedef_import"

func main() {
	// this is OK


	s := template_typedef_import.NewSin()
	s.Get_base_value()
	s.Get_value()
	s.Get_arith_value()
	template_typedef_import.My_func_r(s)
	template_typedef_cplx2.Make_Multiplies_double_double_double_double(s, s)

	z := template_typedef_import.NewCSin()
	z.Get_base_value()
	z.Get_value()
	z.Get_arith_value()
	template_typedef_import.My_func_c(z)
	template_typedef_cplx2.Make_Multiplies_complex_complex_complex_complex(z, z)

	// Here we fail
	d := template_typedef_cplx2.Make_Identity_double()
	template_typedef_import.My_func_r(d)

	c := template_typedef_cplx2.Make_Identity_complex()
	template_typedef_import.My_func_c(c)
}

File: SWIG/Examples/test-suite/python/template_typedef_import_runme.py
import string
from template_typedef_cplx2 import *
from template_typedef_import import *

#
# this is OK
#


s = Sin()
s.get_base_value()
s.get_value()
s.get_arith_value()
my_func_r(s)
make_Multiplies_double_double_double_double(s,s)

z = CSin()
z.get_base_value()
z.get_value()
z.get_arith_value()
my_func_c(z)
make_Multiplies_complex_complex_complex_complex(z,z)

#
# Here we fail
#
try:
  d = make_Identity_double()
  my_func_r(d)
except:
  print d, "is not a RFunction?, but yes it is!" 
  raise RuntimeError

try:
  c = make_Identity_double()
  my_func_c(c)
except:
  print c, "is not a CFunction?, but yes it is!" 
  raise RuntimeError
  



